libs/core/src/lib/list/list-item/list-item.component.ts

```diff
@@ -219,19 +197,9 @@ export class ListItemComponent
         this._listenOnButtonQueryChange();
 
         if (this.linkDirectives && this.linkDirectives.length) {
-            this._listItemRole += ' button';
-        }
-    }
-
-    /** @hidden */
-    ngOnChanges(changes: SimpleChanges): void {
-        if (
-            changes.selectedListItemScreenReaderText ||
-            changes.navigatedListItemScreenReaderText ||
-            changes.navigatableListItemScreenReaderText ||
-            changes.selected
-        ) {
-            this._updateScreenReaderText();
+            this._role = 'link';
+        } else if (this.radio || this.checkbox) {
+            this._role = 'option';
```

Comment:
What if in the runtime, these values change and none of the requirements get fulfilled? It will stay incorrect value.  I suppose we need some change listener

Reply:
I can do this in the contentchild setters